src/Avalonia.Base/Data/Core/Plugins/DataAnnotationsValidationPlugin.cs

```diff
@@ -33,16 +35,16 @@ private sealed class Accessor : DataValidationBase
         {
             private readonly ValidationContext _context;
 
-            public Accessor(WeakReference<object> reference, string name, IPropertyAccessor inner)
+            public Accessor(WeakReference<object?> reference, string name, IPropertyAccessor inner)
                 : base(inner)
             {
-                reference.TryGetTarget(out object target);
+                reference.TryGetTarget(out var target);
 
                 _context = new ValidationContext(target);
                 _context.MemberName = name;
             }
```

Comment:
Unrelated to this PR, but I wonder why we pass weakreference in ctor and make a strong reference right after that.

Reply:
I guess because data binding system is working with weak refs but half of the actual plugins just take a strong ref at some point anyway.

Reply:
Yeah I think the problem is that the data binding system works with null references but `ValidationContext` (which is part of the BCL) doesn't.

Reply:
Oh, I see. Haven't noticed it comes from BCL.